### PR TITLE
Apply merge block checks from fork choice on _block

### LINKED ISSUE #4644: [Merge] Implement updated fork choice onBlock method
### Description
Need to implement the updated fork choice `on_block` function to additionally verify the transition block conditions.

https://github.com/ethereum/consensus-specs/blob/dev/specs/merge/fork-choice.md#updated-fork-choice-handlers

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Checks the terminal total difficulty conditions are met for the "merge block" (first one with a real execution payload).

Also removes `difficulty` from `PowBlock` because we don't use it and its being removed from the spec soon.

## Fixed Issue(s)
fixes #4644

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/forkchoice/ForkChoiceTestExecutor.java [hunk 1]
```diff
@@ -56,7 +56,7 @@
           .put(
               "fork_choice/on_block",
               new ForkChoiceTestExecutor("new_finalized_slot_is_justified_checkpoint_ancestor"))
-          .put("fork_choice/on_merge_block", new ForkChoiceTestExecutor())
+          .put("fork_choice/on_merge_block", IGNORE_TESTS)
```
Comment (ajsutton) [inline eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/forkchoice/ForkChoiceTestExecutor.java:59]: Ignoring these as we need to update the test runner to execute them properly before they'll pass (they should be failing now as they actually require this change).

File: ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoicePayloadExecutorTest.java [hunk 2]
```diff
@@ -71,17 +82,36 @@ void setUp() {
...
+  @Test
+  void optimisticallyExecute_shouldNotExecuteDefaultPayload() {
+    final ForkChoicePayloadExecutor payloadExecutor = createPayloadExecutor();
+    final boolean result = payloadExecutor.optimisticallyExecute(payloadHeader, defaultPayload);
+    verify(executionEngine, never()).executePayload(any());
+    assertThat(result).isTrue();
+  }
+
+  @Test
+  void optimisticallyExecute_shouldValidateMergeBlockWhenThisIsTheMergeBlock() {
```
Comment (rolfyone) [inline ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoicePayloadExecutorTest.java:110]: this seems more like `optimisticallyExecute_shouldCheckParentHashBeforExecutingPayload` or something?

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/merge/helpers/MergeTransitionHelpers.java [hunk 3]
```diff
@@ -13,48 +13,74 @@
...
+    return executionEngine
+        .getPowBlock(parentBlockHash)
+        .thenCompose(
+            maybeParentPowBlock -> {
+              if (maybeParentPowBlock.isEmpty()) {
+                return completedFuture(ExecutePayloadResult.SYNCING);
+              }
+              final PowBlock parentPowBlock = maybeParentPowBlock.get();
+              if (!isBelowTotalDifficulty(parentPowBlock)) {
+                return invalid("Parent PowBlock exceeds terminal total difficulty");
```
Comment (rolfyone) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/merge/helpers/MergeTransitionHelpers.java:91]: isn't it that the parent block hasn't exceeded total difficulty?

Comment (rolfyone) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/merge/helpers/MergeTransitionHelpers.java:91]: actually never mind that i think im wrong..

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**